### ISSUE TITLE
When scsslint ruby gem isn't installed, it shouldn't say that all files are lint free

### ISSUE THREAD

**Issue by willemdewit** [ahmednuaman/grunt-scss-lint#92]
When running the grunt task on a system which hasn't installed the ruby gem, it says that all files are lint free. 

It is very hard to debug and figure out that the ruby gem isn't installed. It should throw an error.

**Comment by ahmednuaman** [comment]
Correct, it _should_ throw an error. Can you tell me more so I can see why the current checks aren't working correctly?

**Comment by willemdewit** [comment]
Ok, I run `gem uninstall scss-lint` and I can reproduce the problem.

When I place the following lines on line 172 of `node_modules/grunt-scss-lint/tasks/lib/scss-lint.js`, (just above `if (err && err.code !== 1 && err.code !== 2 && err.code !== 65) {`)

    console.log(err.code);
    console.log(err);

it outputs:

    $ node.exe C:\Users\wdwit\AppData\Roaming\npm\node_modules\grunt-cli\bin\grunt --gruntfile C:\[SKIPPED]\gruntfile.js scsslint:button
    Running "scsslint:button" (scsslint) task
    Running scss-lint on button
    1
    { [Error: Command failed: 'scss-lint' is not recognized as an internal or external command,
    operable program or batch file.
    ] killed: false, code: 1, signal: null }
    >> 4 files are lint free

Btw, my operating system is Windows 8.

**Comment by tbremer** [comment]
I can't replicate this issue on a Macbook Pro (Retina, Yosemite). With this `ruby -v` and these gems:
```bash
$ ruby -v
ruby 2.1.3p242 (2014-09-19 revision 47630) [x86_64-darwin14.0]

$ gem list

*** LOCAL GEMS ***

bigdecimal (1.2.4)
bourbon (4.0.2)
bundler (1.7.4)
bundler-unload (1.0.2)
celluloid (0.16.0)
celluloid-io (0.16.1)
executable-hooks (1.3.2)
ffi (1.9.6)
gem-wrappers (1.2.7)
hitimes (1.2.2)
io-console (0.4.2)
json (1.8.1)
listen (2.7.11)
minitest (4.7.5)
nio4r (1.0.1)
psych (2.0.5)
rainbow (2.0.0)
rake (10.1.0)
rb-fsevent (0.9.4)
rb-inotify (0.9.5)
rdoc (4.1.0)
rubygems-bundler (1.4.4)
rubygems-update (2.4.2)
rvm (1.11.3.9)
sass (3.4.7, 3.2.19)
test-unit (2.1.3.0)
thor (0.19.1)
timers (4.0.1)
```

![gruntfile](https://cloud.githubusercontent.com/assets/2998575/5724442/3491a944-9b11-11e4-94b8-af79e62cec2b.png)

My GruntFile.js looks like:
```javascript
// I AM USING THE TEST FILES FROM THE MAIN INSTALL
module.exports = function (grunt) {
  grunt.initConfig({

    scsslint: {
      options: {
        config: 'fixtures/.scss-lint-test.yml'
      },
      all: [
        'fixtures/fail.scss',
        'fixtures/pass.scss',
        'fixtures/*.scss'
      ],
      success: [
        'fixtures/pass.scss'
      ]
    }
  });

  grunt.loadNpmTasks('grunt-scss-lint');

  grunt.registerTask('default', ['scsslint'])
};
```

My output is:
```bash
grunt
Running "scsslint:all" (scsslint) task
Running scss-lint on all
>> 1. Please make sure you have ruby installed: `ruby -v`
>> 2. Install the `scss-lint` gem by running:
>> gem update --system && gem install scss-lint
Warning: Task "scsslint:all" failed. Use --force to continue.

Aborted due to warnings.
```

Works as expected when I install the latest scss-lint installed
```
$ scss-lint -v
scss-lint 0.32.0
```

Not saying this doesn't exist, but it might be a Windows issue -- I don't have a Windows box to test on.

**Comment by ahmednuaman** [comment]
Thanks @tbremer; TBH @willemdewit I haven't properly tested it on Windows 8, I do have a VM running Windows 7, but will need a hand with 8. So at least now there's an error, did you not get this before? What were the conditions in which it passed linting but no linting occurred?

**Comment by willemdewit** [comment]
The problem is that the if statement doesn't match so the error is silently passing. 

The if statement: `if (err && err.code !== 1 && err.code !== 2 && err.code !== 65) {` checks if the `err.code !== 1`. But on windows 8 it is 1. Because there is no else statement the program continues.

**Comment by tbremer** [comment]
Well, your example above returns an err.code of 1, which means that there was a warning in your linted files.

To me this implies you have scss-lint installed OR that the error codes on Mac and Windows are different (either in ruby or causes/scsslint).

**Comment by willemdewit** [comment]
I think the latter. The first time I faced this problem I was on a clean install of Windows.

**Comment by ahmednuaman** [comment]
These are the exit codes we have, I guess it needs a bit more finessing, but I certainly am interested exactly what the output was first time round. I'm going to pull down a Windows 8 WM and try it out, is there anymore you can tell me about your install?

**Comment by willemdewit** [comment]
The gems I have installed are the following:

    C:\Windows\system32>gem list
    
    *** LOCAL GEMS ***
    
    bigdecimal (1.2.4)
    io-console (0.4.2)
    json (1.8.1)
    minitest (5.5.0, 4.7.5)
    power_assert (0.2.2)
    psych (2.0.5)
    rainbow (2.0.0)
    rake (10.4.2, 10.1.0)
    rdoc (4.2.0, 4.1.0)
    sass (3.4.9, 3.4.8)
    test-unit (3.0.8, 2.1.5.0)
    win32-api (1.5.2 universal-mingw32)
    windows-api (0.4.3)
    windows-pr (1.2.4)

I have a 64-bit operating system.

**Comment by ahmednuaman** [comment]
So is there anyway you can replicate the initial error?

**Comment by willemdewit** [comment]
By running `gem uninstall scss-lint` and then run the grunt task.

**Comment by ahmednuaman** [comment]
So you get [this](https://github.com/ahmednuaman/grunt-scss-lint/issues/92#issuecomment-69740567) error but not [this](https://github.com/ahmednuaman/grunt-scss-lint/issues/92#issue-54161805) error?

**Comment by willemdewit** [comment]
By default it is **not** throwing an error at all. 

But when I place the following lines on line 172 of node_modules/grunt-scss-lint/tasks/lib/scss-lint.js, (just above `if (err && err.code !== 1 && err.code !== 2 && err.code !== 65) {`)

    console.log(err.code);
    console.log(err);

it shows me an error.code of 1, and the output mentioned [here](https://github.com/ahmednuaman/grunt-scss-lint/issues/92#issuecomment-69740567)

**Comment by ahmednuaman** [comment]
So can I has a pull request?

**Comment by ahmednuaman** [comment]
Do you mind checking now?

**Comment by willemdewit** [comment]
I uninstalled the gem scss-lint and run the grunt task again. It is still giving the same issue.

**Comment by Qazzian** [comment]
I've just had the same problem and did some more checks.
If the command scss-lint is not available then windows returns an error code of 1

**Comment by ahmednuaman** [comment]
Good idea, I've created a branch and am testing: 3f52965

**Comment by nschonni** [comment]
What about taking a page from grunt-contrib-sass https://github.com/gruntjs/grunt-contrib-sass/blob/73546de3cb077feaef5c0ba8a2307c1b2d23c432/tasks/sass.js#L13

**Comment by ahmednuaman** [comment]
All merged for 0.3.8
